src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -528,4 +528,18 @@ public HaxeComponentName getComponentName() {
     }
   };
 
+
+  // FIX: for twice deletion of file in project view (issue #424)
+  @Override
+  public void delete() {
+    final HaxeFile file = (HaxeFile)getContainingFile();
+    if(file != null) {
+      final PsiClass[] classes = file.getClasses();
+      if(classes.length == 1 && classes[0] == this) {
+        file.delete();
+        return;
+      }
+    }
+    super.delete();
```

Comment:
Thank you! You approach is much more considered :+1: 
I've made correction changes. IDE delete/undo functionality has been checked.